lib/plugins/aws/lib/naming.js

```diff
@@ -379,22 +379,24 @@ module.exports = {
   },
 
   // ALB
-  getAlbTargetGroupLogicalId(functionName) {
-    return `${this.getNormalizedFunctionName(functionName)}AlbTargetGroup`;
+  getAlbTargetGroupLogicalId(functionName, albId) {
+    return `${this.getNormalizedFunctionName(functionName)}AlbTargetGroup${albId}`;
   },
-  getAlbTargetGroupNameTagValue(functionName) {
+  getAlbTargetGroupNameTagValue(functionName, albId) {
     return `${
       this.provider.serverless.service.service
-    }-${functionName}-${this.provider.getStage()}`;
+    }-${functionName}-${albId}-${this.provider.getStage()}`;
   },
-  getAlbTargetGroupName(functionName) {
+  getAlbTargetGroupName(functionName, albId) {
     return crypto
       .createHash('md5')
-      .update(this.getAlbTargetGroupNameTagValue(functionName))
+      .update(this.getAlbTargetGroupNameTagValue(functionName, albId))
       .digest('hex');
   },
-  getAlbListenerRuleLogicalId(functionName, idx) {
-    return `${this.getNormalizedFunctionName(functionName)}AlbListenerRule${idx}`;
+  getAlbListenerRuleLogicalId(functionName, priority, albId, listenerId) {
+    return `${this.getNormalizedFunctionName(
```

Comment:
I think change back this function to its original version should do the trick (make the PR non-breaking) but I have to test it to confirm.